libraries/standard/http/include/http_client.h

```diff
@@ -403,19 +439,44 @@ HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * #HTTPRequestHeaders_t.pBuffer.
  *
  * For example, if requesting for the first 1kB of a file the following would be
- * written  "Range: bytes=0-1024\r\n".
+ * written  "Range: bytes=0-1024\r\n\r\n".
+ *
+ * The trailing "\r\n" that denotes the end of the header lines is overwritten, if it
+ * already exists in the buffer.
+ *
+ * @note There are 3 different forms of range specification, determined by the
+ * combination of @a rangeStartOrLastNBytes and @a rangeEnd parameter values:
+ * 1. Request containing both parameters for the byte range [rangeStart, rangeEnd]
+ * where @a rangeStartOrLastNBytes <= @a rangeEnd.
+ * Example request: "Range: bytes=0-1024\r\n" for requesting bytes in the range [0, 1024].
```

Comment:
We should probably change this to 1023 to be consistent with the example change I commented.